punjab/session.py

```diff
@@ -379,11 +379,17 @@ def _pollForId(self, d):
 
 
     def connectEvent(self, xs):
-
         self.version =  self.authenticator.version
         self.xmlstream = xs
         self.xmlstream.rawDataOutFn = self.rawDataOut
         self.xmlstream.rawDataInFn = self.rawDataIn
+        if self.pint.proxy_protocol:
+            local_port = self.xmlstream.transport.getHost().port
```

Comment:
If getHost() is None your already in trouble, that is the connection from the client to you.